lms/djangoapps/grades/progress.py

```diff
@@ -1,159 +1,11 @@
-    # Django translation --> ... --> courseware --> submissions
-    from submissions import api as sub_api  # installed from the edx-submissions repository
-    with outer_atomic():
-        submissions_scores = sub_api.get_scores(
-            unicode(course.id), anonymous_id_for_user(student, course.id)
-        )
-
-    # Check for gated content
-    gated_content = gating_api.get_gated_content(course, student)
-
-    chapters = []
-    locations_to_weighted_scores = {}
-
-    for chapter_key in course_structure.get_children(course_structure.root_block_usage_key):
-        chapter = course_structure[chapter_key]
-        sections = []
-        for section_key in course_structure.get_children(chapter_key):
-            if unicode(section_key) in gated_content:
-                continue
-
-            section = course_structure[section_key]
-
-            graded = getattr(section, 'graded', False)
-            scores = []
-
-            for descendant_key in course_structure.post_order_traversal(
-                    filter_func=possibly_scored,
-                    start_node=section_key,
-            ):
-                descendant = course_structure[descendant_key]
-
-                (correct, total) = get_score(
-                    student,
-                    descendant,
-                    scores_client,
-                    submissions_scores,
-                )
-                if correct is None and total is None:
-                    continue
-
-                weighted_location_score = Score(
-                    correct,
-                    total,
-                    graded,
-                    block_metadata_utils.display_name_with_default_escaped(descendant),
-                    descendant.location
-                )
-
-                scores.append(weighted_location_score)
-                locations_to_weighted_scores[descendant.location] = weighted_location_score
-
-            escaped_section_name = block_metadata_utils.display_name_with_default_escaped(section)
-            section_total, _ = graders.aggregate_scores(scores, escaped_section_name)
-
-            sections.append({
-                'display_name': escaped_section_name,
-                'url_name': block_metadata_utils.url_name_for_block(section),
-                'scores': scores,
-                'section_total': section_total,
-                'format': getattr(section, 'format', ''),
-                'due': getattr(section, 'due', None),
-                'graded': graded,
-            })
-
-        chapters.append({
-            'course': course.display_name_with_default_escaped,
-            'display_name': block_metadata_utils.display_name_with_default_escaped(chapter),
-            'url_name': block_metadata_utils.url_name_for_block(chapter),
-            'sections': sections
-        })
-
-    return ProgressSummary(chapters, locations_to_weighted_scores, course_structure.get_children)
+    return CourseGradeFactory(student).create(course)
```

Comment:
I agree, but I also believe that this will be going away before terribly long as well. In the context of "Progress" and "Course Grades", I could see where a "Progress" summary contains details of all steps, where a "Course Grades" summary has already "summarized" that data.

Let's definitely try to get rid of them in the short-term future, but I think it'll be okay to remain in this stepping-stone work.